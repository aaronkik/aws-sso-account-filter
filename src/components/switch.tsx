import { Switch as _Switch, type SwitchProps as _SwitchProps } from '@headlessui/react';
import { Fragment } from 'react';

interface SwitchProps extends _SwitchProps {
  checked: boolean;
  onChange: (value: boolean) => void;
}

const Switch = (props: SwitchProps) => {
  return (
    <_Switch {...props} checked={props.checked} onChange={props.onChange} as={Fragment}>
      {({ checked }) => (
        <button
          className={`${
            checked ? 'bg-orange-400' : 'bg-slate-400'
          } relative inline-flex h-7 w-11 min-w-11 items-center rounded-sm`}
        >
          <span
            className={`${
              checked ? 'translate-x-6' : 'translate-x-1'
            } inline-block h-5 w-4 transform rounded-sm bg-white transition`}
          />
        </button>
      )}
    </_Switch>
  );
};

export default Switch;
