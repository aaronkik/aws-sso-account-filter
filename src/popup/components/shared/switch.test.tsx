import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, test, vi } from 'vitest';
import { Switch } from './switch';

describe('Switch', () => {
  test('matches snapshot when unchecked', () => {
    const { container } = render(<Switch checked={false} onChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test('matches snapshot when checked', () => {
    const { container } = render(<Switch checked={true} onChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test('calls onChange when clicked', async () => {
    const user = userEvent.setup();
    const onChange = vi.fn();

    render(<Switch checked={false} onChange={onChange} />);
    const switchElement = screen.getByRole('switch');

    await user.click(switchElement);

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(true);
  });
});
