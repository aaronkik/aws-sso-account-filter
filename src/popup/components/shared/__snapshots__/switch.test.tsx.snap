// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch > matches snapshot when checked 1`] = `
<div>
  <button
    aria-checked="true"
    class="bg-orange-400 relative inline-flex h-7 w-11 min-w-11 items-center rounded-sm"
    data-checked=""
    data-headlessui-state="checked"
    id="headlessui-switch-«r1»"
    role="switch"
    tabindex="0"
    type="button"
  >
    <span
      class="translate-x-6 inline-block h-5 w-4 transform rounded-sm bg-white transition"
    />
  </button>
</div>
`;

exports[`Switch > matches snapshot when unchecked 1`] = `
<div>
  <button
    aria-checked="false"
    class="bg-slate-400 relative inline-flex h-7 w-11 min-w-11 items-center rounded-sm"
    data-headlessui-state=""
    id="headlessui-switch-«r0»"
    role="switch"
    tabindex="0"
    type="button"
  >
    <span
      class="translate-x-1 inline-block h-5 w-4 transform rounded-sm bg-white transition"
    />
  </button>
</div>
`;
